SwiftCompilerSources/Sources/Basic/SourceLoc.swift

```diff
@@ -0,0 +1,73 @@
+//===--- SourceLoc.swift - SourceLoc bridiging utilities ------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+public struct SourceLoc {
+  public var pointer: UnsafePointer<UInt8>
```

Comment:
1. This should be a `let`.
2. `pointer` is a bad name. How about `locationInFile`. It needs a comment that this points into a source file, i.e. the pointee is a character in the source file.
3. I'm a bit nervous to expose an UnsafePointer as a public API. Is this really necessary? If not, let's remove the `public`.
4. Why `UInt8` and not `CChar`?

Reply:
1. Made it `let`
2. Renamed to `locationInFile`
3. When C++ code passes `SourceLoc` to a Swift function, that Swift code might want to access the source text from it. 
4. Swift's `String` UTF8 (e.g. `String(decoding:as:UTF8.self)`) uses `UInt8` collection. Since conversion between `unsigned char *` and `char *` is easier in C++, than conversion between `UnsafePointer<UInt8>` and `UnsafePointer<CChar>`, I thought it makes sense to bridge c string pointer as `unsigned char *`